Core2/Onboarding/src/main/java/com/infomaniak/library/onboarding/OnboardingScaffold.kt

```diff
@@ -0,0 +1,152 @@
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.ImageVector.Builder
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun OnboardingScaffold(
+    pagerState: PagerState,
+    onboardingPages: List<OnboardingPage>,
+    bottomContent: @Composable (PaddingValues) -> Unit,
+) {
+    Scaffold { paddingValues ->
+        Column {
+            val startPadding = paddingValues.calculateStartPadding(LocalLayoutDirection.current)
+            val endPadding = paddingValues.calculateEndPadding(LocalLayoutDirection.current)
+
+            HorizontalPager(pagerState, modifier = Modifier.weight(1f)) { pageIndex ->
+                OnboardingPageContent(
+                    page = onboardingPages[pageIndex],
+                    contentPadding = PaddingValues(
+                        top = paddingValues.calculateTopPadding(),
+                        start = startPadding,
+                        end = endPadding,
+                    ),
+                )
+            }
+
+            HorizontalPagerIndicator(
+                modifier = Modifier.padding(PaddingValues(start = startPadding, end = endPadding)),
+                pagerState = pagerState,
+            )
+
+            bottomContent(
+                PaddingValues(
+                    bottom = paddingValues.calculateBottomPadding(),
+                    start = startPadding,
+                    end = endPadding,
+                )
+            )
+        }
+    }
+}
+
+@Composable
+private fun OnboardingPageContent(page: OnboardingPage, contentPadding: PaddingValues) {
+    Box(
+        modifier = Modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center,
+    ) {
+        Image(
+            imageVector = page.background,
+            contentDescription = null,
+            modifier = Modifier.fillMaxSize(),
+            contentScale = ContentScale.FillBounds,
+        )
+        Column(
+            modifier = Modifier
+                .fillMaxHeight()
+                .verticalScroll(rememberScrollState())
+                .padding(contentPadding),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.SpaceEvenly,
+        ) {
+            with(page) {
+                illustration()
+                text()
+            }
+        }
+    }
+}
+
+data class OnboardingPage(
+    val background: ImageVector,
+    val illustration: @Composable () -> Unit,
+    val text: @Composable () -> Unit,
+)
+
+@Preview
+@Preview(device = "spec:parent=pixel_5,orientation=landscape")
```

Comment:
The previews from SwissTransfer are not accessible from this module unfortunately :/

Reply:
Ultimately we definitely should, but instead of moving half the project to Core2 I tried limiting the modifications to what felt like the most important parts. 

Do we deem the previews important enough and I move them already here?

Reply:
The Preview's code is small enough and generic enough to be moved now, I guess.
It could be in another PR though.

Reply:
We create a card to do this task later 👍